biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -1175,38 +1207,58 @@ public void run() {
 		return systemBundle;
 	}
 
-	protected void deleteFiles(File wd) {
-		IO.delete(wd);
+	private Framework createClassic(Framework systemBundle, ClassLoader loader, Map<String, String> configuration) {
+		FrameworkFactory factory = getMetaInfService(loader, FrameworkFactory.class);
+		if (factory != null) {
+			systemBundle = factory.newFramework(configuration);
+			trace("framework instance %s", systemBundle);
+		} else
+			trace("framework factory not found");
+		return systemBundle;
 	}
 
-	/**
-	 * Try to get the stupid service interface ...
-	 *
-	 * @throws IOException
+	/*
+	 * Attempts to create an OSGi Connect. This allows a third party bundle on
+	 * the -runpath to control the class loading strategy
+	 * @param loader the class loader of this launcher
+	 * @param configuration the framework configuration
+	 * @return null or a connect framework
 	 */
-	private List<String> getMetaInfServices(ClassLoader loader, String factory) throws IOException {
-		if (loader == null)
-			loader = getClass().getClassLoader();
-
-		Enumeration<URL> e = loader.getResources("META-INF/services/" + factory);
-		List<String> factories = new ArrayList<>();
-
-		while (e.hasMoreElements()) {
-			URL url = e.nextElement();
-			trace("found META-INF/services in %s", url);
-
-			try (BufferedReader rdr = IO.reader(url.openStream(), UTF_8)) {
-				String line;
-				while ((line = rdr.readLine()) != null) {
-					trace("%s", line);
-					line = line.trim();
-					if (!line.startsWith("#") && line.length() > 0) {
-						factories.add(line);
-					}
+	private Framework createConnect(ClassLoader loader, Map<String, String> configuration) {
```

Comment:
How do we test this actually works? We have tests now for normal OSGi framework launching (both explicit and implicit tests).

Reply:
I want it in there so we can experiment. Once the frameworks are available we can have tests. You can only activate the code by putting a Module Connector services on your -runpath, which seems a good safeguard this is not activated accidentally..